sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -197,7 +197,9 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
           Some(partitionSchema))
 
         val logicalRelation = cached.getOrElse {
-          val sizeInBytes = relation.stats.sizeInBytes.toLong
+          val defaultSizeInBytes = sparkSession.sessionState.conf.defaultSizeInBytes
```

Comment:
For write path, the stats for HiveTable is not calculated as InsertIntoTable's table filed is not added in the children.

So `org.apache.spark.sql.hive.DetermineTableStats` skip this table

Reply:
cc @wangyum do you think this is fixable?

Reply:
Maybe we can be recalculate stats by:
https://github.com/apache/spark/blob/485ae6d1818e8756a86da38d6aefc8f1dbde49c2/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala#L74-L77

This is test case:
https://github.com/apache/spark/blob/485ae6d1818e8756a86da38d6aefc8f1dbde49c2/sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala#L1488-L1500

Reply:
Fixed by DetermineTableStats handling InsertIntoTable(HiveTableRelation, ...)